### PR TITLE
[crypto] document some modules

### PR DESCRIPTION
**Pull request by mimoo**
documenting random things as I look at different parts of the code

### CODE REVIEW

File: src/lib/pickles/step_verifier.ml [hunk 1]
```diff
@@ -70,6 +70,7 @@ struct
       (Inner_curve.to_field_elements g2)
     |> Boolean.all
 
+  (** Absorbs `t` (of type `ty` into the sponge `sponge`. *)
```
Comment (mrmr1993) [inline src/lib/pickles/step_verifier.ml:73]: Don't use backticks in OCaml doc-comments, use square brackets.

File: src/lib/pickles/ro.ml [hunk 2]
```diff
@@ -1,27 +1,47 @@
+(** doc *)
+
 open Core_kernel
 open Backend
 open Pickles_types
 open Import
 
-let bits_random_oracle =
+(** hash `s` with blake2s and return `length` bits of the digest *)
```
Comment (mrmr1993) [inline src/lib/pickles/ro.ml:8]: Replace backticks with square brackets

File: src/lib/pickles/wrap_hack.ml [hunk 3]
```diff
@@ -23,7 +23,9 @@ open Pickles_types
 
 module Padded_length = Nat.N2
 
-(* Pad up to length 2 by preprending dummy values. *)
+(** Pads a vector with `dummy` element to make it of size 2.
```
Comment (mrmr1993) [inline src/lib/pickles/wrap_hack.ml:26]: Replace backticks with square brackets

File: src/lib/pickles/wrap_hack.ml [hunk 4]
```diff
@@ -23,7 +23,9 @@ open Pickles_types
...
+    Warning: panics if the vector has more than one element.
```
Comment (mrmr1993) [inline src/lib/pickles/wrap_hack.ml:27]: This isn't true. This `throws an exception if the vector has more than 2 elements`

File: src/lib/pickles/step_verifier.ml (hunk 1 above)
Comment (mimoo) [reply src/lib/pickles/step_verifier.ml:73]: I thought square brackets were links to types that existed, whereas here they are arguments to the function

**Comment by mimoo** [comment]
ping @mrmr1993

**Review by mitschabaude — CHANGES_REQUESTED** [review]
hey, this checks out snarkyjs to an old commit, is this a mistake?

**Comment by mimoo** [comment]
The joy of git submodules

**Comment by mimoo** [comment]
I really need to figure out what's a one-liner git command to revert all the git submodules back to their versions in develop

**Review by mitschabaude — APPROVED** [review]
sorry for not approving for so long, this fell under my radar somehow :/

**Comment by mimoo** [comment]
ping @mrmr1993 @imeckler

File: src/lib/pickles/step_verifier.ml [hunk 5]
```diff
@@ -70,6 +70,7 @@ struct
       (Inner_curve.to_field_elements g2)
     |> Boolean.all
 
+  (** Absorbs [t] (of type [ty] into the sponge [sponge]. *)
```
Comment (mrmr1993) [inline src/lib/pickles/step_verifier.ml:73]: Missing closing paren

File: src/lib/pickles/wrap_hack.ml [hunk 6]
```diff
@@ -23,7 +23,9 @@ open Pickles_types
 
 module Padded_length = Nat.N2
 
-(* Pad up to length 2 by preprending dummy values. *)
+(** Pads a vector with [dummy] element to make it of size 2.
+    Warning: panics if the vector has more than two elements.
```
Comment (mrmr1993) [inline src/lib/pickles/wrap_hack.ml:27]: OCaml vernacular is `raises`

File: src/lib/pickles/wrap_hack.ml [hunk 7]
```diff
@@ -89,12 +91,13 @@ module Checked = struct
...
       chalss
 
+  (** pads `commitments` up to length 2 *)
   let pad_commitments (commitments : _ Vector.t) =
-    pad_vector
-      ~dummy:
-        (Tuple_lib.Double.map ~f:Impls.Step.Field.constant
-           (Lazy.force Dummy.Ipa.Wrap.sg) )
-      commitments
+    (* use dummy sg to pad the commitments vector to a length 2 *)
```
Comment (imeckler) [inline src/lib/pickles/wrap_hack.ml:96]: nit: should say `use dummy sg to pad the commitments vector to length 2` (no "a")

**Review by imeckler — APPROVED** [review]

**Comment by mrmr1993** [comment]
Stale. Closing.

**Pull request closed without merging**